### PR TITLE
[Fix-8624][UI-Next][V1.0.0-Alpha] Rectify the issue to display the timing form when clicking the timing button in the process definition page.

### LINKED ISSUE #8624: [Bug][UI-Next][V1.0.0-Alpha] click timing error in process definition page
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What you expected to happen

click successfully.

### How to reproduce

above.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by calvinjiang**
## Purpose of the pull request
This PR will close #8624 .

## Verify this pull request
This change added tests and can be verified as follows:
  - *Manually verified the change by testing locally.* 
image

### CODE REVIEW

File: dolphinscheduler-ui-next/src/views/projects/workflow/definition/components/timing-modal.tsx [hunk 1]
```diff
@@ -188,8 +188,8 @@ export default defineComponent({
       () => props.row,
       () => {
         timingState.timingForm.startEndTime = [
-          new Date(props.row.startTime),
-          new Date(props.row.endTime)
+          props.row.startTime? new Date(props.row.startTime) : new Date(),
```
Comment (devosend) [inline dolphinscheduler-ui-next/src/views/projects/workflow/definition/components/timing-modal.tsx]: You can use `props.row.crontab` to determine whether timeForm init is needed. Beause timeForm.crontab and other's needs to be initialized.

Comment (calvinjiang) [reply dolphinscheduler-ui-next/src/views/projects/workflow/definition/components/timing-modal.tsx]: No problem. I'll improve that.

**Comment by devosend** [comment]
LGTM

**Review by songjianet — APPROVED** [review]
LGTM

**Pull request merged**